azurerm/helpers/validate/public_ip.go

```diff
@@ -0,0 +1,14 @@
+package validate
+
+import (
+	"fmt"
+	"regexp"
+)
+
+func PublicIpDomainNameLabel(v interface{}, k string) (ws []string, errors []error) {
+	value := v.(string)
+	if !regexp.MustCompile(`^[a-z][a-z0-9-]{1,61}[a-z0-9]$`).MatchString(value) {
+		errors = append(errors, fmt.Errorf("Domain must contain only lowercase alphanumeric characters, numbers and hyphens. It must start with a letter and end only with a number or letter"))
```

Comment:
the argument `k` here is the name of the field name being validated - so we could make this:

```suggestion
		errors = append(errors, fmt.Errorf("%s must contain only lowercase alphanumeric characters, numbers and hyphens. It must start with a letter and end only with a number or letter", k))
```